lib/tramway/configs/entity.rb

```diff
@@ -1,24 +1,44 @@
 # frozen_string_literal: true
 
+require 'tramway/configs/entities/route'
+
 module Tramway
   module Configs
-    # Tramway is entity based framework
-    #
-    class Entity
-      attr_reader :name
+    # Tramway is an entity-based framework
+    class Entity < Dry::Struct
+      attribute :name, Types::Coercible::String
+      attribute? :route, Tramway::Configs::Entities::Route
 
-      def initialize(name:)
-        @name = name.to_s
+      def routes
+        OpenStruct.new index: Rails.application.routes.url_helpers.public_send(route_helper_method)
       end
 
-      def routes
-        underscored_name = name.parameterize.pluralize.underscore
+      def human_name
+        options = if model_class.present?
+                    model_name = model_class.model_name.human
+                    { single: model_name, plural: model_name.pluralize }
+                  else
+                    { single: name.capitalize, plural: name.pluralize.capitalize }
+                  end
+        OpenStruct.new(**options)
+      end
+
+      private
 
-        OpenStruct.new index: Rails.application.routes.url_helpers.public_send("#{underscored_name}_path")
+      def model_class
+        name.camelize.constantize
+      rescue StandardError
+        nil
       end
 
-      def human_name
-        OpenStruct.new single: name.capitalize, plural: name.pluralize.capitalize
+      def route_helper_method
```

Comment:
I've created method `routes` in the `Tramway::Config::Entity` because there will be logic about additional routes that can be defined in Tramway Config for each entity.